### DISCUSSION TITLE
Example RemoteApp in Docker Container?

### DISCUSSION THREAD

**Discussion by slominskir** [FreeRDP/FreeRDP#8644] [Q&A] [answered]
It appears the RDP RemoteApp feature may be an alternative to X11 when accommodating a GUI app launched from within a Docker container.   Are there any examples of this?    I noticed a [similar setup](https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps) with WSL2 in which launching a GUI app leverages [FreeRDP RemoteApp](https://github.com/microsoft/wslg#wslg-architecture-overview) to display a Linux window on a Windows desktop.   You can even launch a Docker container from within WSL2 that can [pass the display](https://github.com/microsoft/wslg/blob/main/samples/container/Containers.md#containerizing-gui-applications-with-wslg) to the Windows Desktop.   I can't find an example of a more generic RemoteApp solution though such as launching Docker from a Windows CMD terminal (outside of WSL2) or even launching Docker from a Linux host.  In both of these later scenarios I assume the user would need to manually invoke a remote app command on the host to connect to a FreeRDP server running in a Linux container.

**Comment by akallabeth** [comment, marked as answer]
not possible, as neighter `gnome-remote-desktop`, `ogon` nor `xrdp` support the remote app mode server side. (that is, at this point in time)

the windows `WSL2` thing only uses `FreeRDP` for the metadata communication (e.g. movements, resizes, ...), the windows are passed through directly
